docs/_build/html/_sources/tiltfile_concepts.md.txt

```diff
@@ -0,0 +1,106 @@
+# Tiltfile Concepts
+
+This doc describes concepts in the Tiltfile, expanding on the [Tutorial](tutorial.html). Unlike the [API Reference](api.html), it groups functions by themes and explains why you'd choose to use a function.
+
+## Execution Model
+`Tiltfile`s are written in [Starlark](https://github.com/bazelbuild/starlark), a dialect of Python. Tilt executes the `Tiltfile` on startup.
+
+Functions like `k8s_yaml` and `docker_build` register information. At the end of the execution, Tilt uses the resulting configuration. In addition to the final configuration, Tilt records file accesses. Tilt listens to these files, and reexecutes when one changes (but not on every source file change).
+
+Because Tilt uses a programming language, you can configure it with familiar constructs like loops, functions, arrays, etc. This makes Tilt more extensible than a configuration that requires hard-coding all possible options up-front.
```

Comment:
```suggestion
Because your Tiltfile is written in a programming language, you can configure it with familiar constructs like loops, functions, arrays, etc. This makes Tilt more extensible than a configuration that requires hard-coding all possible options up-front.
```

Reply:
WDYT?